### PR TITLE
Fix: Clear cache and service worker data on reload

### LINKED ISSUE #2928: Force a cache clean and bypass service workers when possible
Windows 8.1
Canary 78 and Chrome 76
Webhint Extension 0.0.7

I'm using the suggested Apache .htaccess configuration, yet this hint is thrown for the webmanifest. It's once again appending the charset to the content-type; reason unknown.

> hint 1: 'content-type' header value should not contain 'charset=utf-8'.
> https://thomasbrodhead.com/favicons/site.20190904080058.webmanifest
> Content-Type: application/manifest+json; charset=utf-8

Is this related to Issue #2927?

### PR DESCRIPTION
**Pull request by molant**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- ~~Added/Updated related documentation.~~
- ~~Added/Updated related tests.~~

## Short description of the change(s)

Explicitly delete the cache and any service worker data before reloading. 

Fix #2928

Bundles for testing:

[chromium 2019-09-09](https://github.com/webhintio/hint/files/3592792/webhint-0.0.7.zip)
[firefox 2019-09-09](https://github.com/webhintio/hint/files/3592794/webhint-0.0.7.zip)

### CODE REVIEW

**Comment by molant** [comment]
Yes, I need to update tests. I wanted to validate it was working for @bmt-systems but it looks like more work is needed.

**Comment by molant** [comment]
I've updated this PR and right now I'm just ignoring requests coming from service workers. Seems to be more reliable.

**Comment by molant** [comment]
Also the bundles are updated

File: packages/extension-browser/src/devtools/utils/network.ts [hunk 1]
```diff
@@ -79,10 +79,45 @@ const generateFetchEnd = async (entry: chrome.devtools.network.Request) => {
     hops.delete(url);
 };
 
+/**
+ * Checks if a request is comming from a Service|Web worker using the
+ * `referer` header to identify it.
```
Comment (antross) [inline packages/extension-browser/src/devtools/utils/network.ts]: Probably good to include *why* we want to ignore requests from workers somewhere in the comments. Otherwise looks good.

Comment (molant) [reply packages/extension-browser/src/devtools/utils/network.ts]: This is explained in line 118

**Comment by molant** [comment]
This is ready for another review.
I've updated the bundles in the first comment as well.

File: packages/extension-browser/src/content-script/connector.ts [hunk 2]
```diff
@@ -147,7 +161,7 @@ export default class WebExtensionConnector implements IConnector {
         this.setFetchElement(event);
         const type = this.setFetchType(event);
 
-        await this._engine.emitAsync(`fetch::end::${type}` as 'fetch::end::*', event);
+        await this.handleFetch(`fetch::end::${type}` as 'fetch::end::*', { fetchEnd: event });
```
Comment (antross) [inline packages/extension-browser/src/content-script/connector.ts]: Thinking about this some more I realized it only fixes the Chrome code path for the `Fetcher` class. When running in Firefox we return the response of a `fetch` call initiated by `Fetcher` directly. This means `setFetchElement` and `setFetchType` never get a chance to run in that context so we're still missing `element` and `mediaType` data.

Comment (antross) [reply packages/extension-browser/src/content-script/connector.ts]: Arguably `element` isn't as relevant here since these were triggered by a `fetchContent` call to begin with. This means the caller should have the `element` already so we don't need to wait for it here. And `mediaType` is derived from the headers so we can call that anytime.

TL;DR - I think we can roll back this particular change, add the `mediaType` from within the `Fetcher` class itself, and require callers of `fetchContent` to set the `element`.

**Comment by molant** [comment]
😒

________________________________

File: packages/extension-browser/src/content-script/connector.ts (hunk 2 above)
Comment (molant) [reply packages/extension-browser/src/content-script/connector.ts]: Updated. The `element` is already added in `parser-manifest` so no need to re-add it.

File: packages/extension-browser/src/content-script/fetcher.ts [hunk 3]
```diff
@@ -108,6 +110,8 @@ export class Fetcher {
     private resolveFetch(event: FetchEnd) {
```
Comment (antross) [inline packages/extension-browser/src/content-script/fetcher.ts]: To make this work in the Firefox code-path, we should replace the following above (around line 70) with a call to this `resolveFetch` method instead.

```diff
-                if (this._fetches.has(target) && response.type === 'basic') {
+                if (response.type === 'basic') {
                    // Firefox code-path (can't use other path as fetch doesn't trigger onRequestFinished).
                    const data = await toNetworkData(target, headers, response);

-                    resolve(data);
-                    this._fetches.delete(target);
+                    this.resolveFetch(data);
                } else {
```

Comment (antross) [reply packages/extension-browser/src/content-script/fetcher.ts]: Resolving as we discussed this data is already set by the `toNetworkData` helper.
